src/component/completed_list.rs

```diff
@@ -43,6 +44,24 @@ impl CompletedList {
             "Restores the selected task",
         )]
     }
+
+    pub fn restore_task(&mut self, app: &mut App) {
+        if app.task_store.completed_tasks.is_empty() {
+            return;
+        }
+        app.task_store.tasks.push(Task::from_completed_task(
```

Comment:
Sorry, I should have made myself more clear. This allows removal of the sort further down.
```suggestion
        app.task_store.add_task(Task::from_completed_task(
```